scripts/generate_psa_wrappers.py

```diff
@@ -5,254 +5,15 @@
-            if self._parameter_should_be_copied(function.name,
-                                                function.arguments[param.index].name))
-        self._write_poison_buffer_parameters(out, buffer_parameters, True)
-        super()._write_function_call(out, function, argument_names)
-        self._write_poison_buffer_parameters(out, buffer_parameters, False)
-
-    def _write_prologue(self, out: typing_util.Writable, header: bool) -> None:
-        super()._write_prologue(out, header)
-        out.write("""
-#if {}
-
-#include <psa/crypto.h>
-
-#include <test/memory.h>
-#include <test/psa_crypto_helpers.h>
-#include <test/psa_test_wrappers.h>
-"""
-                  .format(self._CPP_GUARDS))
-
-    def _write_epilogue(self, out: typing_util.Writable, header: bool) -> None:
-        out.write("""
-#endif /* {} */
-"""
-                  .format(self._CPP_GUARDS))
-        super()._write_epilogue(out, header)
-
-
-class PSALoggingWrapperGenerator(PSAWrapperGenerator, c_wrapper_generator.Logging):
-    """Generate a C source file containing wrapper functions that log PSA Crypto API calls."""
-
-    def __init__(self, stream: str) -> None:
-        super().__init__()
-        self.set_stream(stream)
-
-    _PRINTF_TYPE_CAST = c_wrapper_generator.Logging._PRINTF_TYPE_CAST.copy()
-    _PRINTF_TYPE_CAST.update({
-        'mbedtls_svc_key_id_t': 'unsigned',
-        'psa_algorithm_t': 'unsigned',
-        'psa_drv_slot_number_t': 'unsigned long long',
-        'psa_key_derivation_step_t': 'int',
-        'psa_key_id_t': 'unsigned',
-        'psa_key_slot_number_t': 'unsigned long long',
-        'psa_key_lifetime_t': 'unsigned',
-        'psa_key_type_t': 'unsigned',
-        'psa_key_usage_flags_t': 'unsigned',
-        'psa_pake_role_t': 'int',
-        'psa_pake_step_t': 'int',
-        'psa_status_t': 'int',
-    })
-
-    def _printf_parameters(self, typ: str, var: str) -> Tuple[str, List[str]]:
-        if typ.startswith('const '):
-            typ = typ[6:]
-        if typ == 'uint8_t *':
-            # Skip buffers
-            return '', []
-        if typ.endswith('operation_t *'):
-            return '', []
-        if typ in self._PAKE_STUFF:
-            return '', []
-        if typ == 'psa_key_attributes_t *':
-            return (var + '={id=%u, lifetime=0x%08x, type=0x%08x, bits=%u, alg=%08x, usage=%08x}',
-                    ['(unsigned) psa_get_key_{}({})'.format(field, var)
-                     for field in ['id', 'lifetime', 'type', 'bits', 'algorithm', 'usage_flags']])
-        return super()._printf_parameters(typ, var)
-
+from mbedtls_framework.code_wrapper.psa_test_wrapper import PSATestWrapper, PSALoggingTestWrapper
 
 DEFAULT_C_OUTPUT_FILE_NAME = 'tests/src/psa_test_wrappers.c'
 DEFAULT_H_OUTPUT_FILE_NAME = 'tests/include/test/psa_test_wrappers.h'
 
-def main() -> None:
-    parser = argparse.ArgumentParser(description=globals()['__doc__'])
+if __name__ == '__main__':
```

Comment:
There are arguably some reasons to have a separate main() method but I would argue it not Pythonic. 

The whole scope of the automatic `__name__=="main"` check is to ensure that everything under it is not imported or executed other than by calling the module directly. Adding the requirement for main code to be importable, is a bit counter intuitive.

In this specific file all this code does is it import `argparser`, get the user input and call the appropriate libary methods to do what the user calling the script directly, wants to do. I fail to see how that code would/should be shared and exposed to other modules.

In addition `main()` would be evaulated on an import statement causing argparser to be needlesly imported.

Reply:
> There are arguably some reasons to have a separate main() method but I would argue it not Pythonic.

The official documentation says otherwise.

> Putting as few statements as possible in the block below if `__name__ == '__main__'` can improve code clarity and correctness. Most often, a function named main encapsulates the program’s primary behavior:

> In addition main() would be evaulated on an import statement causing argparser to be needlesly imported.

Uh? No, `main` is not evaluated on import. If the module is imported, either way, the same code gets executed. All the imports are at the top and unconditional, which is generally recommended unless there's a strong reason not to do it, which there isn't here.

Reply:
That won't be changed. There is practically no reason for invoking argparser on a separate main function. 

We should not be aiming to complicate code for the sake of it.

`main()` has purpose in complicated multithreaded programs, like async servers, games etc. Not for a script passing two parameters onto single line of code.

Reply:
Sorry, but that's a blocker. A `main` function is [the recommended way to structure non-toy programs](https://docs.python.org/3/library/__main__.html#idiomatic-usage) and has objective advantages as explained above.

Even pylint will reject your code because `parser` and other variables are global variables and it will want them to use the naming convention for global variables. Don't disable the warning: fix the code.

Reply:
This is not addressed as of 03bfc58000f7e6ff35a0f4a97821619c054ee275.

Reply:
This is the definiton of a toy program. We are just passing around some parameters from command line into a class. 

I will ammend it as you requested, but I am not confident that we should be letting the tools dictate the complexity of our code.

Reply:
This is not a toy program: it's hundreds of lines of code divided in several files. The “tools” in question are the language itself. Pylint isn't always right, but in this case, it is correctly pointing out some dodgy code.